lib/src/combine_latest.dart

```diff
@@ -193,8 +193,8 @@ extension CombineLatest<T> on Stream<T> {
       void handleData(int index, T data) {
         latestData[index] = data;
         hasEmitted.add(index);
-        if (hasEmitted.length == allStreams.length) {
-          controller.add(List.from(latestData));
+        if (hasEmitted.length == allStreams.length || !waitAll) {
+          controller.add(List.from(latestData.where((data) => data != null)));
```

Comment:
If `T` is a nullable type this won't do what we want.

If we want to stop supporting null unsafe code we might be able to use `whereType<T>()` here. If we wanted to keep supporting null unsafe code we'd need some more complexity. I think it's probably fine to bump our SDK constraint and drop support for null unsafe code.

Reply:
This would still be an issue in null safe code wouldn't it? When dealing with generics I thought the only way to unambiguously use `null` as a sentinel would be to put an `Object` bound on `T`.

Reply:
Decision is your's but I think null unsafe code is quite outdated now

Reply:
> This would still be an issue in null safe code wouldn't it?

Yes, that's a good point - it would make it look like some false `null` were added in the source streams when the type happens to be nullable.

We probably need to use `hasEmitted`

```dart
        if (hasEmitted.length == allStreams.length) {
          controller.add(List.from(latestData));
        } else if (!waitAll) {
          var emittingIndexes = hasEmitted.toList()..sort();
          controller
              .add([for (var index in emittingIndexes) latestData[index] as T]);
        }
```